agreement/proposal.go

```diff
@@ -56,8 +56,9 @@ type unauthenticatedProposal struct {
 	bookkeeping.Block
 	SeedProof crypto.VrfProof `codec:"sdpf"`
 
-	OriginalPeriod   period         `codec:"oper"`
-	OriginalProposer basics.Address `codec:"oprop"`
+	OriginalPeriod   period           `codec:"oper"`
+	OriginalProposer basics.Address   `codec:"oprop"`
+	ctx              *context.Context //TODO(yg): set to background?
```

Comment:
We needed to use a pointer in order to be able to set the context in the proposal verification thread, and then be able to use it in the network sending thread

Reply:
Sure, but Context is an interface, so using context.Context is what you want here.

Reply:
Can you explain why Context being an interface makes a difference? This was made a pointer since the unauthenticatedProposal was not being passed by reference to several places. Thus when. we set `up.ctx = someContext` the update wouldn't persist to the other unauthenticatedProposal